### PR TITLE
Add Windows Build Pipeline

### PR DESCRIPTION
**Pull request by jw-msft**
- Uses Microsoft-Hosted vmImage
- Triggered by feature branch for now (will be triggered by develop too once merged there)
- Builds and runs unit tests
- Publish test results
- Publish bin and lib artifacts (including .PDB symbols)

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -28,7 +28,9 @@ Param(
     [string]$LogLib = 'zlog',
     # Log folder to use for DU logs
     # TODO(JeffMill): Change this when folder structure determined.
-    [string]$LogDir = '/var/log/adu'
+    [string]$LogDir = '/var/log/adu',
+    # Running in pipeline?
+    [switch]$Pipeline
```
Comment (JeffMill) [inline scripts/build.ps1:33]: i'd prefer "-NoPrompt" (currently i trigger off of "-Clean") or perhaps checking if we're running in an interactive console or not

[Environment]::UserInteractive maybe?

File: scripts/build.ps1 [hunk 2]
```diff
@@ -249,7 +251,11 @@ if ($BuildDocumentation) {
 # Set default log dir if not specified.
 $runtime_dir = "$BuildOutputPath/bin"
 $library_dir = "$BuildOutputPath/lib"
+
 $cmake_bin = 'cmake.exe'
+$cmake_cmd = Get-Command cmake.exe
```
Comment (JeffMill) [inline scripts/build.ps1:256]: `-CommandType Application`

File: scripts/build.ps1 [hunk 3]
```diff
@@ -249,7 +251,11 @@ if ($BuildDocumentation) {
...
+$cmake_bin = $cmake_cmd.Path
+Write-Verbose ('Using $cmake_cmd.Path')
```
Comment (JeffMill) [inline scripts/build.ps1:258]: parens not needed.

argument wont be expanded as you're not using ".
did you mean write-verbose ('using {0}' -f $make_cmd.Path) ?

**Review by JeffMill — APPROVED** [review]
approved with comments

**Review by Nox-MSFT — APPROVED** [review]

File: azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+
+pr:
+    branches:
+        include:
+            - feature/wiot-s1
+            - user/jw-msft/win-pipeline
```
[the remark this answers is not in the document]
Comment (jw-msft) [reply azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml:6]: This is so I could see this PR itself running compliance check. Will remove with a PR after its merged.

File: azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    - task: PublishPipelineArtifact@1
+      inputs:
+        targetPath: 'out/Debug/bin/Debug'
+        artifact: 'bin'
+        publishLocation: 'pipeline'
+
+    - task: PublishPipelineArtifact@1
+      inputs:
+        targetPath: 'out/Debug/lib/Debug'
```
[the remark this answers is not in the document]
Comment (jw-msft) [reply azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml:70]: Yes, I'm working on adding the ARM64 cross-compile GeneratorPlatform you added, and that one will be Release just to mix it up a bit.

File: scripts/build.ps1 (hunk 3 above)
Comment (jw-msft) [reply scripts/build.ps1:258]: will address in next PR

File: azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+
+pr:
+    branches:
+        include:
+            - feature/wiot-s1
+            - user/jw-msft/win-pipeline
+    paths:
+        exclude:
+            - docs/*
+            - README.md
```
[the remark this answers is not in the document]
Comment (jw-msft) [reply azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml:10]: Only would include vcpkg.json in the root methinks

[the remark this answers is not in the document]
Comment (jw-msft) [reply azurepipelines/build/native/adu-windows-msvc2022-amd64-build.yml:10]: based on ado docs it doesn't seem like there's a wildcard for root level that does not also exclude all the dirs, but I'll see if I can improve this a bit

**Pull request merged**